src/query-builder/transformer/PlainObjectToDatabaseEntityTransformer.ts

```diff
@@ -129,16 +130,14 @@ export class PlainObjectToDatabaseEntityTransformer {
         fillLoadMap(plainObject, metadata)
         // load all entities and store them in the load map
         await Promise.all(
-            loadMap.groupByTargetIds().map((targetWithIds) => {
-                // todo: fix type hinting
-                return this.manager
-                    .findByIds<ObjectLiteral>(
-                        targetWithIds.target as any,
-                        targetWithIds.ids,
-                    )
-                    .then((entities) =>
-                        loadMap.fillEntities(targetWithIds.target, entities),
-                    )
+            loadMap.groupByTargetIds().map(async (targetWithIds) => {
+                const foundEntities = await this.manager.findBy<ObjectLiteral>(
+                    targetWithIds.target,
+                    {
+                        id: In([...targetWithIds.ids]),
```

Comment:
shouldn't we look for the primary key instead of just "id" ?

Reply:
Well I just refactored the findByIds implementation, not the logic. This is the default. I think this is just a conveniance function for the often used column `id` as unique identifier. For all other cases you will have to use `findBy` instead.